tutorial/markdown/aspnet/quickstart-csharp-aspnet-travel-sample/quickstart-csharp-aspnet-travel-sample.md

```diff
@@ -0,0 +1,646 @@
+
+            string query;
+            if (!string.IsNullOrEmpty(country))
+            {
+                query = $@"SELECT airline.callsign,
+                            airline.country,
+                            airline.iata,
+                            airline.icao,
+                            airline.name
+                            FROM airline AS airline
+                            WHERE lower(airline.country) = $country
+                            ORDER BY airline.name
+                            LIMIT $limit
+                            OFFSET $offset";
+                queryParameters.Parameter("country", country.ToLower());
+            }
+            else
+            {
+                query = $@"SELECT airline.callsign,
+                            airline.country,
+                            airline.iata,
+                            airline.icao,
+                            airline.name
+                            FROM airline AS airline
+                            ORDER BY airline.name
+                            LIMIT $limit
+                            OFFSET $offset";
+            }
+
+            var results = await inventoryScope.QueryAsync<Airline>(query, queryParameters);
+            var items = await results.Rows.ToListAsync();
+
+            return items.Count == 0 ? Results.NotFound() : Results.Ok(items);
+        }
+        else
+        {
+            return Results.Problem("Scope not found");
+        }
+    }
+    catch (Exception ex)
+    {
+        return Results.Problem(ex.Message);
+    }
+})
+```
+
+### Direct Connections
+
+This endpoint fetches the airports that can be reached directly from the specified source airport code. This also uses a SQL++ query to fetch the results similar to the List Airport endpoint.
+
+Let us look at the query used here:
+
+```sql
+SELECT distinct (route.destinationairport)
+FROM airport as airport
+JOIN route as route on route.sourceairport = airport.faa
+WHERE airport.faa = $airport and route.stops = 0
+ORDER BY route.destinationairport
+LIMIT $limit
+OFFSET $offset
+```
+
+We are fetching the direct connections by joining the airport collection with the route collection and filtering based on the source airport specified by the user and by routes with no stops.
+
+## Running The Tests
+
+We have defined unit tests using the [xunit](https://xunit.net/) nuget package for all the API end points. The unit tests use the same database configuration as the application. For the unit tests, we perform the operation using the API and confirm the results by checking the documents in the database. For example, to check the creation of the document by the API, we would call the API to create the document and then read the same document from the database and compare them. After the tests, the documents are cleaned up by calling the DELETE endpoint
+
+To run the standard integration tests, use the following commands:
```

Comment:
These are integration tests. Can you change the references to unit tests

Reply:
Done.